R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
 
 #' Retrieve all copy number segments from the GAMBL database that overlap with a single genomic coordinate range.
 #'
-#' @param chromosome The chromosome you are restricting to.
-#' @param qstart Start coordinate of the range you are restricting to.
-#' @param qend End coordinate of the range you are restricting to.
-#' @param region Region formatted like chrX:1234-5678 instead of specifying chromosome, start and end separately.
-#' @param with_chr_prefix Prepend all chromosome names with chr (required by some downstream analyses).
+#' @param region Region formatted like chrX:1234-5678 or X:1234-56789.
+#' @param chromosome The chromosome you are restricting to. Required parameter if region is not specified.
+#' @param qstart Start coordinate of the range you are restricting to. Required parameter if region is not specified.
+#' @param qend End coordinate of the range you are restricting to. Required parameter if region is not specified.
+#' @param projection Selected genome projection for returned Cn segments.
+#' @param this_seq_type Seq type for returned Cn segments. Currently, only genome is supported. Capture samples will be added once processed through CNV protocols.
+#' @param remove_chr_prefix Prepend all chromosome names with chr (required by some downstream analyses). Default is FALSE. 
 #' @param streamlined Return a basic rather than full MAF format.
-#' @param from_flatfile Set to FALSE by default.
+#' @param from_flatfile Set to TRUE by default.
 #'
-#' @return A data frame containing all the MAF data columns (one row per mutation).
+#' @return A data frame with CN segments for the specified region.
 #' @export
-#' @import tidyverse DBI RMariaDB
+#' @import DBI RMariaDB
+#' @depend tidyverse
 #'
 #' @examples
-#' #basic usage
-#' my_segments = get_cn_segments(region="chr8:128,723,128-128,774,067")
-#' #specifying chromosome, start and end individually
-#' my_segments = get_cn_segments(chromosome="8",qstart=128723128,qend=128774067)
-#' #Asking for chromosome names to have a chr prefix (default is un-prefixed)
-#' prefixed_segments = get_cn_segments(chromosome ="12",qstart = 122456912, qend = 122464036, with_chr_prefix = TRUE)
-#'
-get_cn_segments = function(chromosome = "",
+#' #Example using chromosome, qstart and qend parameters:
+#' segments_region_grch37 = get_cn_segments(chromosome = "chr8",
+#'                                          qstart = 128723128,
+#'                                          qend = 128774067,
+#'                                          projection = "grch37", 
```

Comment:
Projection and genome here are same as default, so maybe let's drop them from example to show the "minimum usage"? The next example then shows how to modify different projection and I think the flatfiles are not available for capture anyways

Reply:
Indeed, I'll go over the examples and ensure parameters called with default values are not shown.